client/src/modules/stock/exit/exit.js

```diff
@@ -299,6 +316,46 @@ function StockExitController(
       .catch(Notify.handleError);
   }
 
+  // on lot select
+  function onLotSelect(row) {
+    if (!row.lot || !row.lot.uuid) { return; }
+
+    checkValidity();
+
+    refreshSelectedLotsList();
+  }
+
+  // update the list of selected lots
+  function refreshSelectedLotsList() {
+    vm.selectedLots = vm.stockForm.store.data
+      .filter(item => item.lot && item.lot.uuid)
+      .map(item => item.lot.uuid);
+  }
+
+  // detect the presence of duplicated lots
+  function hasDuplicatedLots() {
+    refreshSelectedLotsList();
+
+    let doublonDetected = false;
+    for (let i = 0; i < vm.selectedLots.length; i++) {
+      let found = 0;
```

Comment:
As an alternative to the above suggestion, you can also make this easier to read, but maybe at a loss of performance.  It uses [lastIndexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/lastIndexOf):

```js
let doubleIndex;
const doublonDetected = vm.selectedLots.some((lot, idx) => {
  const hasDoubles = vm.selectedLots.lastIndexOf(lot) !== idx;
  if (hasDoubles) { doubleIndex = idx; }
  return hasDoubles;
});
```

Reply:
6 lines instead of 18!